test/issues/general/test_3821.test

```diff
@@ -0,0 +1,24 @@
+# name: test/issues/general/test_3821.test
+# description: Issue 3821:Setting default_collation=NOCASE modifies values of GROUP BY queries
+# group: [general]
+
+#statement ok
+#PRAGMA enable_verification
+
+statement ok
+CREATE TABLE t AS (SELECT 'hello' AS x UNION ALL SELECT 'WORLD' AS x);
+
+query I rowsort
+SELECT x FROM t GROUP BY 1;
+----
+WORLD
+hello
+
+statement ok
+PRAGMA default_collation=NOCASE;
+
+query I rowsort
+SELECT x FROM t GROUP BY 1;
+----
+WORLD
```

Comment:
Can we add a second `WoRlD` to the test table to ensure the collation is still doing the collation?